app/src/main/java/cloudcity/CloudCityOMNTApplication.java

```diff
@@ -21,6 +23,9 @@ public void onCreate() {
             @Override
             public void onIperf3TestCompleted(MetricsPOJO metrics) {
                 android.util.Log.wtf(TAG, "One iperf3 cycle is completed! received metrics: "+metrics);
+                DataProvider dp = GlobalVars.getInstance().get_dp();
+                boolean iperf3SendingResult = CloudCityUtil.sendIperf3Data(metrics, dp.getLocation());
+                android.util.Log.d(TAG, "sending metrics result: "+iperf3SendingResult);
             }
```

Comment:
would be really great to move this out from the Application class, but for some reason things don't seem to work/fire at all when moved to MainActivity or our MainActivityExtensions so lets leave them here for the demo...

Reply:
sure, and it does kinda work for a while, but then just stops for some reason so... I'll leave it here.

Reply:
resolving.